wapitiCore/parsers/swagger.py

```diff
@@ -0,0 +1,293 @@
+import json
+import sys
+import urllib.parse
+from prance import ResolvingParser, ValidationError
+from prance.util.url import ResolutionError
+
+from wapitiCore.net import Request
+from wapitiCore.main.log import logging
+
+class Swagger():
+    AUTOFILL_VALUES = {
+        "file": ("pix.gif", b"GIF89a", "image/gif"),
+        "integer": "1337",
+        "string": "default",
+        "time": "13:37",
+        "url": "https://wapiti-scanner.github.io/",
+        "boolean": "true",
+        "object": {},
+    }
+
+
+    def __init__(self, uri: str = None) -> None:
+        if uri:
+            try:
+                swaggerFile = ResolvingParser(uri, backend='openapi-spec-validator',
+                                                strict=False, recursion_limit=5).specification
+            except ValidationError as e:
+                logging.error("[-] Error: Swagger file is not valid : " + str(e.args[0]) +
+                                ". See https://swagger.io/specification/ for more information.")
+                sys.exit(1)
+            except AssertionError:
+                logging.error("[-] Error: File not found")
+                sys.exit(1)
+            except ResolutionError:
+                logging.error("[-] Error: Unable to resolve the swagger file")
+                sys.exit(1)
+        else:
+            logging.error("[-] Error: No URL or file")
+            sys.exit(1)
+
+        self.routes = self._get_routes(swaggerFile, uri)
+
+
+    @staticmethod
+    def _get_base_url(swaggerFile: dict, url: str) -> str:
+        try:
+            if 'schemes' not in swaggerFile:
+                # get http or https from url
+                swaggerFile['schemes'] = urllib.parse.urlparse(url).scheme
+            if 'host' not in swaggerFile:
+                if url:
+                    swaggerFile['host'] = urllib.parse.urlparse(url).hostname
+                else:
+                    swaggerFile['host'] = ""
+            elif swaggerFile['host'] == "localhost" and url:
+                swaggerFile['host'] = urllib.parse.urlparse(url).hostname
+            if 'basePath' not in swaggerFile:
+                swaggerFile['basePath'] = ""
+            if 'https' in swaggerFile['schemes']:
+                return 'https://' + swaggerFile['host'] + swaggerFile['basePath']
+            return 'http://' + swaggerFile['host'] + swaggerFile['basePath']
+        except ValueError as e:
+            logging.error("[-] Error: Swagger file is not valid : " + str(e) +
+                          ". See https://swagger.io/specification/ for more information.")
+            sys.exit(1)
+
+
+    @staticmethod
+    def _check_properties(model_name):
+        if "properties" in model_name:
+            return model_name['properties']
+        if "additionalProperties" in model_name:
+            return model_name['additionalProperties']
+        return model_name
+
+
+    def _parse_object(self, model_name):
```

Comment:
As someone who is missing Swagger knowledge :upside_down_face:  , it would be great to have a description about what that function is doing